.github/workflows/test.yml

```diff
@@ -73,6 +73,7 @@ jobs:
         output-dir: wheelhouse
       env:
         CIBW_ARCHS_MACOS: x86_64 universal2 arm64
+        CIBW_SKIP: pp*-macosx_arm64
```

Comment:
Hmmm. on further consideration this does seem a little like a footgun - many people will have cross-compilation written into their configs, which will fail when this is added. Can we perhaps provide a softer failure on macOS when pp*_arm64 is erroneously selected on x86_64?

Reply:
e.g. when we attempt to build pp-arm64 on x86_64, we print-

```
warning: pp38-macosx_arm64 was selected but can't be built on this architecture (x86_64). Deselect these builds using CIBW_SKIP=pp*-macosx_arm64 to remove this warning.
```

I realise this is a special case, but I expect many people to hit a failed build otherwise.

On this other hand, we could say that we'd rather error out for consistency. But since our example configs have included this since arm64 support was added, it's gonna hit a lot of people.

Reply:
We could even downgrade that from a warning to a notice, since there's no real problem with not having skipped pp*_arm64.

Reply:
Actually, since we don't support pypy cross compilation, I don't even know if we need a notice - it should simply not be a selectable option on a non-native platfrom, IMO, unless we add cross compilation.

Reply:
I went with @henryiii proposal to filter out this configuration. Tests were updated to check this, let's see CI results.

Reply:
I'd still be in favour of a notice, since it's the only build identifier that's not selectable in this way.